[Bug] Crash on startup if no WebView

Just tried running AnkiDroid on an Android 8 emulator. On me to triage, could be a bad collection

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki, PID: 3215
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.ichi2.libanki.Deck.getString(java.lang.String)' on a null object reference
        at com.ichi2.libanki.Decks.select(Decks.java:1090)
        at com.ichi2.libanki.Decks.current(Decks.java:1080)
        at com.ichi2.anki.DeckPicker.__renderPage(DeckPicker.java:2511)
        at com.ichi2.anki.DeckPicker$UpdateDeckListListener.actualOnPostExecute(DeckPicker.java:2403)
        at com.ichi2.anki.DeckPicker$UpdateDeckListListener.actualOnPostExecute(DeckPicker.java:2374)
        at com.ichi2.async.TaskListenerWithContext.onPostExecute(TaskListenerWithContext.java:66)
        at com.ichi2.async.CollectionTask.onPostExecute(CollectionTask.java:225)
        at android.os.AsyncTask.finish(AsyncTask.java:695)
        at android.os.AsyncTask.-wrap1(Unknown Source:0)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

`mDecks.getDid(1) == null`

Issue with no WebView Renderer. Seems to create a corrupt collection

```
07/08 18:32:31: Launching 'AnkiDroid' on 26 (Orea) - 8.0 No Google.
App restart successful without requiring a re-install.
$ adb shell am start -n "com.ichi2.anki/com.ichi2.anki.IntentHandler" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -D
Waiting for application to come online: com.ichi2.anki.tests | com.ichi2.anki
Waiting for application to come online: com.ichi2.anki.tests | com.ichi2.anki
Connected to process 3345 on device '26_Orea_-_8.0_No_Google [emulator-5554]'.
Waiting for application to come online: com.ichi2.anki.tests | com.ichi2.anki
Connecting to com.ichi2.anki
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
I/zygote: Not late-enabling -Xcheck:jni (already on)
W/zygote: Unexpected CPU variant for X86 using defaults: x86
W/ActivityThread: Application com.ichi2.anki is waiting for the debugger on port 8100...
I/System.out: Sending WAIT chunk
I/zygote: Debugger is active
I/System.out: Debugger has connected
I/System.out: waiting for debugger to settle...
Connected to the target VM, address: 'localhost:51994', transport: 'socket'
I/chatty: uid=10065(u0_a65) com.ichi2.anki identical 11 lines
I/System.out: waiting for debugger to settle...
I/System.out: debugger has settled (1426)
I/ACRA: ACRA is disabled for com.ichi2.anki, initializing...
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
I/zygote:     at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
I/zygote:     at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
I/zygote:     at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
I/zygote:     at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<androidx.webkit.internal.WebViewRenderProcessClientFrameworkAdapter>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/webkit/WebViewRenderProcessClient;
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
    Caused by: java.lang.ClassNotFoundException: Didn't find class "android.webkit.WebViewRenderProcessClient" on path: DexPathList[[zip file "/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk"],nativeLibraryDirectories=[/data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/lib/x86, /data/app/com.ichi2.anki-uF-K44CKctVR0Y7KQ9UuAg==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
        at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
        at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
        at android.content.pm.PackageInfo androidx.webkit.WebViewCompat.getCurrentWebViewPackage(android.content.Context) (WebViewCompat.java:353)
        at java.util.HashMap com.ichi2.anki.AnkiDroidApp.fetchWebViewInformation() (AnkiDroidApp.java:718)
        at void com.ichi2.anki.AnkiDroidApp.setAcraConfigBuilder(org.acra.config.CoreConfigurationBuilder) (AnkiDroidApp.java:250)
        at void com.ichi2.anki.AnkiDroidApp.setDebugACRAConfig(android.content.SharedPreferences) (AnkiDroidApp.java:521)
        at void com.ichi2.anki.AnkiDroidApp.onCreate() (AnkiDroidApp.java:290)
        at void android.app.Instrumentation.callApplicationOnCreate(android.app.Application) (Instrumentation.java:1118)
        at void android.app.ActivityThread.handleBindApplication(android.app.ActivityThread$AppBindData) (ActivityThread.java:5791)
        at void android.app.ActivityThread.-wrap1(android.app.ActivityThread, android.app.ActivityThread$AppBindData) (ActivityThread.java:-1)
        at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1661)
        at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:105)
        at void android.os.Looper.loop() (Looper.java:164)
        at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6541)
        at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
        at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:240)
        at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:767)
W/AnkiDroidApp: Could not get WebView package information
W/AnkiDroidApp: Could not get WebView package information
D/AnkiDroid: Startup - Application Start
I/UsageAnalytics: initialize()
D/UsageAnalytics: App tracking id 'tid' = UA-125800786-1
E/WebViewFactory: Chromium WebView package does not exist
    android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getWebViewContextAndSetProvider(WebViewFactory.java:334)
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:398)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211)
        at android.webkit.WebSettings.getDefaultUserAgent(WebSettings.java:1259)
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.setAndroidRequestParameters(UsageAnalytics.java:340)
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.access$100(UsageAnalytics.java:312)
        at com.ichi2.anki.analytics.UsageAnalytics.initialize(UsageAnalytics.java:73)
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:310)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
W/UsageAnalytics$AndroidDefaultRequest: android.util.AndroidRuntimeException: android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:438)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211)
        at android.webkit.WebSettings.getDefaultUserAgent(WebSettings.java:1259)
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.setAndroidRequestParameters(UsageAnalytics.java:340)
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.access$100(UsageAnalytics.java:312)
        at com.ichi2.anki.analytics.UsageAnalytics.initialize(UsageAnalytics.java:73)
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:310)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getWebViewContextAndSetProvider(WebViewFactory.java:334)
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:398)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211) 
        at android.webkit.WebSettings.getDefaultUserAgent(WebSettings.java:1259) 
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.setAndroidRequestParameters(UsageAnalytics.java:340) 
        at com.ichi2.anki.analytics.UsageAnalytics$AndroidDefaultRequest.access$100(UsageAnalytics.java:312) 
        at com.ichi2.anki.analytics.UsageAnalytics.initialize(UsageAnalytics.java:73) 
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:310) 
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791) 
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6541) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
D/NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: true
I/GoogleAnalyticsImpl: Not participating in analytics sample (sample percentage vs random: 10 18)
I/UsageAnalytics: setOptIn(): from false to false
I/GoogleAnalyticsImpl: Not participating in analytics sample (sample percentage vs random: 10 87)
D/UsageAnalytics: setOptIn() optIn / sAnalytics.config().enabled(): false/false
I/UsageAnalytics: setDryRun(): true, warning dryRun is experimental
I/CompatV26: Creating notification channel with id/name: General Notifications/AnkiDroid
I/CompatV26: Creating notification channel with id/name: Synchronization/Synchronization
I/CompatV26: Creating notification channel with id/name: Global Reminders/Cards due
I/CompatV26: Creating notification channel with id/name: Deck Reminders/Reminders
E/WebViewFactory: Chromium WebView package does not exist
    android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getWebViewContextAndSetProvider(WebViewFactory.java:334)
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:398)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211)
        at android.webkit.CookieManager.getInstance(CookieManager.java:39)
        at android.webkit.CookieManager.setAcceptFileSchemeCookies(CookieManager.java:265)
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:331)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
D/UsageAnalytics: sendAnalyticsException() description/fatal: android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed/false
V/ACRA: ACRA is disabled. Report not sent.
E/AnkiDroidApp: setAcceptFileSchemeCookies
    android.util.AndroidRuntimeException: android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:438)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211)
        at android.webkit.CookieManager.getInstance(CookieManager.java:39)
        at android.webkit.CookieManager.setAcceptFileSchemeCookies(CookieManager.java:265)
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:331)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed
        at android.webkit.WebViewFactory.getWebViewContextAndSetProvider(WebViewFactory.java:334)
        at android.webkit.WebViewFactory.getProviderClass(WebViewFactory.java:398)
        at android.webkit.WebViewFactory.getProvider(WebViewFactory.java:211) 
        at android.webkit.CookieManager.getInstance(CookieManager.java:39) 
        at android.webkit.CookieManager.setAcceptFileSchemeCookies(CookieManager.java:265) 
        at com.ichi2.anki.AnkiDroidApp.onCreate(AnkiDroidApp.java:331) 
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791) 
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6541) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
D/IntentHandler: onCreate()
V/IntentHandler: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 cmp=com.ichi2.anki/.IntentHandler }
D/IntentHandler: onCreate() performing default action
I/IntentHandler: Launching DeckPicker
D/AnkiDroidApp: AnkiDroidApp::getLanguageConfig - setting locale to en_US
D/DeckPicker: onCreate()
I/AnkiActivity: AnkiActivity::onCreate - DeckPicker
I/zygote: Do partial code cache collection, code=29KB, data=24KB
    After code cache collection, code=27KB, data=24KB
    Increasing code cache capacity to 128KB
D/OpenGLRenderer: HWUI GL Pipeline
W/CollectionHelper: Failed to query version
    java.lang.NullPointerException: Attempt to invoke virtual method 'int com.ichi2.libanki.Collection.queryVer()' on a null object reference
        at com.ichi2.anki.CollectionHelper.getDatabaseVersion(CollectionHelper.java:507)
        at com.ichi2.anki.CollectionHelper.isFutureAnkiDroidVersion(CollectionHelper.java:492)
        at com.ichi2.anki.InitialActivity.isFutureAnkiDroidVersion(InitialActivity.java:72)
        at com.ichi2.anki.InitialActivity.getStartupFailureType(InitialActivity.java:50)
        at com.ichi2.anki.DeckPicker.handleStartup(DeckPicker.java:558)
        at com.ichi2.anki.DeckPicker.onCreate(DeckPicker.java:467)
        at android.app.Activity.performCreate(Activity.java:6975)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
D/DB: Database /storage/emulated/0/AnkiDroid/collection.anki2 closed = true
I/CollectionHelper: Begin openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
D/RustSupportSQLiteOpenHelper: createRustSupportSQLiteDatabase
I/BackendV1Impl: Opening rust backend. Server: false. Langs: 'en', path: 
I/BackendV1Impl: Opening Collection: '/storage/emulated/0/AnkiDroid/collection.anki2' '' '' ''
D/BackendV1Impl: Rust: fullQueryProto PRAGMA synchronous = 2
D/BackendV1Impl: Rust: fullQueryProto SELECT ver FROM col
D/BackendV1Impl: cancelCurrentProtoQuery
    Rust: fullQueryProto PRAGMA temp_store = memory
    cancelCurrentProtoQuery
I/Collection: Opening Collection Log
D/Collection: [1625765559] Collection.java:<init>(): /storage/emulated/0/AnkiDroid/collection.anki2,  2.16alpha4
D/BackendV1Impl: Rust: fullQueryProto SELECT crt, mod, scm, dty, usn, ls, conf, dconf, tags FROM col
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: Rust: fullQueryProto SELECT substr(decks, ?, ?) FROM col
D/BackendV1Impl: cancelCurrentProtoQuery
D/Collection: [1625765559] Sched.java:_updateCutoff(): 0,  1625799600
I/CollectionHelper: End openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
I/DeckPicker: Displaying database error
I/zygote: Do partial code cache collection, code=57KB, data=60KB
I/zygote: After code cache collection, code=57KB, data=60KB
    Increasing code cache capacity to 256KB
I/zygote: Do full code cache collection, code=57KB, data=78KB
    After code cache collection, code=12KB, data=28KB
D/DeckPicker: No DeckPicker background preference
I/AnkiActivity: AnkiActivity::onStart - DeckPicker
D/DeckPicker: onResume()
I/AnkiActivity: AnkiActivity::onResume - DeckPicker
D/UsageAnalytics: sendAnalyticsScreenView(): DeckPicker
D/UsageAnalytics: getOptIn() status: false
D/DialogHandler: Reading persistent message
D/DeckPicker$UpdateDeckListListener: Refreshing deck list
D/DeckPicker$UpdateDeckListListener: Refreshing deck list
D/UsageAnalytics: sendAnalyticsScreenView(): DatabaseErrorDialog
    getOptIn() status: false
D/CollectionTask$LoadDeck: doInBackgroundLoadDeckCounts
I/Choreographer: Skipped 45 frames!  The application may be doing too much work on its main thread.
D/CollectionTask$LoadDeckCounts: doInBackgroundLoadDeckCounts
D/BackendV1Impl: Rust: fullQueryProto update cards set did = 1 where did not in ()
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: Rust: fullQueryProto SELECT substr(models, ?, ?) FROM col
D/BackendV1Impl: cancelCurrentProtoQuery
D/: HostConnection::get() New Host Connection established 0x8b1be080, tid 3386
I/OpenGLRenderer: Initialized EGL, version 1.4
D/OpenGLRenderer: Swap behavior 1
W/OpenGLRenderer: Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
D/OpenGLRenderer: Swap behavior 0
D/EGL_emulation: eglCreateContext: 0x9a662860: maj 3 min 1 rcv 4
D/EGL_emulation: eglMakeCurrent: 0x9a662860: ver 3 1 (tinfo 0xa670b2b0)
E/eglCodecCommon: glUtilsParamSize: unknow param 0x000082da
E/eglCodecCommon: glUtilsParamSize: unknow param 0x000082da
I/DeckPicker$UpdateDeckListListener: Updating deck list UI
D/BackendV1Impl: Rust: fullQueryProto select avg(case when type = 0 then case when ease > 1 then 1.0 else 0.0 end else null end) as newRate, avg(case when type = 0 then time else null end) as newTime, avg(case when type in (1, 3) then case when ease > 1 then 1.0 else 0.0 end else null end) as revRate, avg(case when type in (1, 3) then time else null end) as revTime, avg(case when type = 2 then case when ease > 1 then 1.0 else 0.0 end else null end) as relrnRate, avg(case when type = 2 then time else null end) as relrnTime from revlog where id > 1624935600000
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: Rust: fullQueryProto SELECT count() FROM cards
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: Rust: fullQueryProto SELECT count() FROM cards
D/BackendV1Impl: cancelCurrentProtoQuery
D/BackendV1Impl: Rust: fullQueryProto SELECT count() FROM cards
D/BackendV1Impl: cancelCurrentProtoQuery
Disconnected from the target VM, address: 'localhost:51994', transport: 'socket'
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java
@@ -547,15 +547,14 @@ public class DeckPicker extends NavigationDrawerActivity implements
      * */
     public void handleStartup() {
         if (Permissions.hasStorageAccessPermission(this)) {
-            boolean colOpen = firstCollectionOpen();
-            if (colOpen) {
+            StartupFailure failure = InitialActivity.getStartupFailureType(this);
+            if (failure == null) {
                 // Show any necessary dialogs (e.g. changelog, special messages, etc)
                 SharedPreferences sharedPrefs = AnkiDroidApp.getSharedPrefs(this);
                 showStartupScreensAndDialogs(sharedPrefs, 0);
                 mStartupError = false;
             } else {
                 // Show error dialogs
-                StartupFailure failure = InitialActivity.getStartupFailureType(this);
                 handleStartupFailure(failure);
                 mStartupError = true;
             }
@@ -590,6 +589,15 @@ public class DeckPicker extends NavigationDrawerActivity implements
                 // This has a callback to continue with handleStartup
                 InitialActivity.downgradeBackend(this);
                 break;
+            case WEBVIEW_FAILED:
+                new MaterialDialog.Builder(this)
+                        .title(R.string.ankidroid_init_failed_webview_title)
+                        .content(getString(R.string.ankidroid_init_failed_webview, AnkiDroidApp.getWebViewErrorMessage()))
+                        .positiveText(R.string.close)
+                        .onPositive((d, w) -> exit())
+                        .cancelable(false)
+                        .show();
+                break;
             case DB_ERROR:
             default:
                 displayDatabaseFailure();
@@ -623,25 +631,6 @@ public class DeckPicker extends NavigationDrawerActivity implements
             view.setBackground(drawable);
             return true;
         }
-    }
-
-    /**
-     * Try to open the Collection for the first time
-     * @return whether or not we were successful
-     */
-    private boolean firstCollectionOpen() {
-        if (AnkiDroidApp.webViewFailedToLoad()) {
-            new MaterialDialog.Builder(this)
-                    .title(R.string.ankidroid_init_failed_webview_title)
-                    .content(getString(R.string.ankidroid_init_failed_webview, AnkiDroidApp.getWebViewErrorMessage()))
-                    .positiveText(R.string.close)
-                    .onPositive((d, w) -> exit())
-                    .cancelable(false)
-                    .show();
-            return false;
-        }
-
-        return CollectionHelper.getInstance().getColSafe(this) != null;
     }
 
     public void requestStoragePermission() {

--- a/AnkiDroid/src/main/java/com/ichi2/anki/InitialActivity.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/InitialActivity.java
@@ -28,7 +28,7 @@ import net.ankiweb.rsdroid.RustBackendFailedException;
 import java.lang.ref.WeakReference;
 
 import androidx.annotation.CheckResult;
-import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import timber.log.Timber;
 
 /** Utilities for launching the first activity (currently the DeckPicker) */
@@ -38,9 +38,21 @@ public class InitialActivity {
 
     }
 
-    @NonNull
+    /** Returns null on success */
+    @Nullable
     @CheckResult
     public static StartupFailure getStartupFailureType(Context context) {
+
+        // A WebView failure means that we skip `AnkiDroidApp`, and therefore haven't loaded the collection
+        if (AnkiDroidApp.webViewFailedToLoad()) {
+            return StartupFailure.WEBVIEW_FAILED;
+        }
+
+        // If we're OK, return null
+        if (CollectionHelper.getInstance().getColSafe(context) != null) {
+            return null;
+        }
+
         if (!AnkiDroidApp.isSdCardMounted()) {
             return StartupFailure.SD_CARD_NOT_MOUNTED;
         } else if (!CollectionHelper.isCurrentAnkiDroidDirAccessible(context)) {
@@ -174,5 +186,7 @@ public class InitialActivity {
         DATABASE_DOWNGRADE_REQUIRED,
         DB_ERROR,
         DATABASE_LOCKED,
+        WEBVIEW_FAILED,
+        ;
     }
 }

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java
@@ -32,6 +32,7 @@ import com.ichi2.utils.JSONException;
 import com.ichi2.utils.JSONObject;
 
 import java.io.File;
+import java.io.FileNotFoundException;
 import java.util.ArrayList;
 import java.util.Arrays;
 
@@ -57,6 +58,9 @@ public class Storage {
     /** Helper method for when the collection can't be opened */
     public static int getDatabaseVersion(String path) throws UnknownDatabaseVersionException {
         try {
+            if (!new File(path).exists()) {
+                throw new UnknownDatabaseVersionException(new FileNotFoundException(path));
+            }
             DB db = new DB(path);
             int result = db.queryScalar("SELECT ver FROM col");
             db.close();

